.gdbinit

```diff
@@ -563,41 +640,67 @@ define printzn
 
 	printf "[%p] %s", $znode, $optype
 
+	# IS_CONST
 	if $znode->op_type == 1
 		printf ": "
 		____printzv &$znode->u.constant 0
 	end
+
+	# IS_TMP_VAR
 	if $znode->op_type == 2
 		printf ": "
 		set $tvar = (union _temp_variable *)((char *)$eg.current_execute_data->Ts + $znode->u.var)
 		____printzv ((union _temp_variable *)$tvar)->tmp_var 0
 	end
+
+	# IS_VAR
 	if $znode->op_type == 4
 		printf ": "
 		set $tvar = (union _temp_variable *)((char *)$eg.current_execute_data->Ts + $znode->u.var)
 		____printzv *$tvar->var.ptr_ptr 0
 	end
+
+	# IS_UNUSED
 	if $znode->op_type == 8
 		printf "\n"
 	end
 end
 
 document printzn
 	print type and content of znode.
-	usage: printzn &opline->op1
+	Usage: printzn (znode*)pznode
+end
+
+# print znode_op information
+# $arg0: znode_op*
+define ____print_znode_op
+	set $op = $arg0
+	printf "[%p]\n", $op
+	printf "  constant=[%d] ", $op->constant
+	printf "var=[%d] ", $op->var
```

Comment:
This is a union, so this macro doesn't really make sense.

Reply:
Yes. I will remove this macro.

Reply:
I see. Dumping all these values still doesn't make much sense, as they are always the same. It would make more sense dump the op_type and the corresponding value (always selecting `->var` should be fine).

Reply:
Agree. I will update the code accordingly.

Reply:
Now the `printzops` functions output looks like this
>(gdb) printzops 
op1     type: IS_CONST  => [0x5555586bed60] var=[-160]
op2     type: IS_UNUSED => [0x5555586bed64] var=[0]
result  type: IS_VAR    => [0x5555586bed68] var=[112]

Reply:
I'd drop the `var=` prefix, as the values are stored in different places, not all as stack variables.

Reply:
That makes sense. How about?
>(gdb) printzops
op1 type: IS_CONST => [0x5555586bed60] value=[-160]
op2 type: IS_UNUSED => [0x5555586bed64] value=[0]
result type: IS_VAR => [0x5555586bed68] value=[112]

Reply:
Looks good! Thanks!